packages/browser/src/eventbuilder.ts

```diff
@@ -0,0 +1,88 @@
+  addExceptionTypeValue,
+  isDOMError,
+  isDOMException,
+  isError,
+  isErrorEvent,
+  isEvent,
+  isPlainObject,
+} from '@sentry/utils';
+
+import { eventFromPlainObject, eventFromStacktrace, prepareFramesForEvent } from './parsers';
+import { computeStackTrace } from './tracekit';
+
+/** JSDoc */
+export function eventFromUnknownInput(exception: unknown, syntheticException?: Error, rejection?: boolean): Event {
+  let event: Event;
+
+  if (isErrorEvent(exception as ErrorEvent) && (exception as ErrorEvent).error) {
+    // If it is an ErrorEvent with `error` property, extract it to get actual Error
+    const errorEvent = exception as ErrorEvent;
+    exception = errorEvent.error; // tslint:disable-line:no-parameter-reassignment
+    event = eventFromStacktrace(computeStackTrace(exception as Error));
+    return event;
+  }
+  if (isDOMError(exception as DOMError) || isDOMException(exception as DOMException)) {
+    // If it is a DOMError or DOMException (which are legacy APIs, but still supported in some browsers)
+    // then we just extract the name and message, as they don't provide anything else
+    // https://developer.mozilla.org/en-US/docs/Web/API/DOMError
+    // https://developer.mozilla.org/en-US/docs/Web/API/DOMException
+    const domException = exception as DOMException;
+    const name = domException.name || (isDOMError(domException) ? 'DOMError' : 'DOMException');
+    const message = domException.message ? `${name}: ${domException.message}` : name;
+
+    event = eventFromString(message);
+    addExceptionTypeValue(event, message);
+    return event;
+  }
+  if (isError(exception as Error)) {
+    // we have a real Error object, do nothing
+    event = eventFromStacktrace(computeStackTrace(exception as Error));
+    return event;
+  }
+  if (isPlainObject(exception) || isEvent(exception)) {
+    // If it is plain Object or Event, serialize it manually and extract options
+    // This will allow us to group events based on top-level keys
+    // which is much better than creating new group when any key/value change
+    const objectException = exception as {};
+    event = eventFromPlainObject(objectException, syntheticException, rejection);
+    addExceptionMechanism(event, {
+      synthetic: true,
+    });
+    return event;
+  }
+
+  // If none of previous checks were valid, then it means that
+  // it's not a DOMError/DOMException
+  // it's not a plain Object
+  // it's not a valid ErrorEvent (one with an error property)
+  // it's not an Error
+  // So bail out and capture it as a simple message:
+  event = eventFromString(exception as string, syntheticException);
+  addExceptionTypeValue(event, `${exception}`, undefined);
+  addExceptionMechanism(event, {
+    synthetic: true,
+  });
+
+  return event;
+}
+
+/** JSDoc */
+export function eventFromString(input: string, syntheticException?: Error): Event {
+  const event: Event = {
+    message: input,
+  };
+
+  // TODO: Only if `this._options.attachStacktrace`
```

Comment:
I think this still needs to be done.
Also, this is a possible candidate for snapshot testing.